src/entry.js

```diff
@@ -85,7 +85,24 @@ class Entry {
    */
   static toMultihash (ipfs, entry) {
     if (!ipfs) throw IpfsNotDefinedError()
-    const data = Buffer.from(JSON.stringify(entry))
+    if (!(entry.id && entry.clock && entry.next && entry.payload && (entry.v === 0 || entry.v > 0))) {
```

Comment:
There's [Entry.isEntry()](https://github.com/thiagodelgado111/ipfs-log/blob/c9032f57ce73071198472a311692d9289f4f7b39/src/entry.js#L145) that is particularly for this kind of check, let's use that one.

If you want, we can add the `v` check into it. Use `entry.v >= 0` rather than `(entry.v === 0 || entry.v > 0)` to check for a positive number.

Reply:
Like I said in https://github.com/orbitdb/ipfs-log/pull/139#discussion_r203447810, `isEntry` expects the object to have a hash already, so if we wanna ensure the object passed in as argument has no `.hash` property, we would have to create a different method that doesn't require that. I agree to change it to `v >= 0` though

Reply:
Understood and you're absolutely correct.

Having thought about this a bit more, let's just go with what you have here. Let's keep it simple :) 

It'd be great to clean this up a bit though with:
- Put the `entry.id && entry.clock && entry.next && entry.payload && (entry.v === 0 || entry.v > 0)` into a function (in the scope of `toMultihash`, doesn't need to be a class-level function) and 
- re-factor the `(entry.v === 0 || entry.v > 0)` as discussed earlier

Both of these are more cosmetic than functional. Does that make sense?

Reply:
Hey, thanks for taking the time to look at it! It makes sense, yes! I've applied the changes you mentioned, updated es5/dist folder and pushed again – please notice I've amended the build artifact commit.